logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/ClientRepository.kt

```diff
@@ -61,11 +69,22 @@ class ClientDataSource(
     override suspend fun registerMLSClient(clientId: ClientId, publicKey: ByteArray): Either<CoreFailure, Unit> = suspending {
         clientRemoteRepository.registerMLSClient(clientId, publicKey.encodeBase64())
     }
-  
+
     override suspend fun saveNewClients(userId: UserId, clients: List<ClientId>): Either<CoreFailure, Unit> =
         userMapper.toUserIdPersistence(userId).let { userEntity ->
             clients.map { ClientEntity(userEntity, it.value) }.let { clientEntityList ->
                 wrapStorageRequest { clientDAO.insertClients(clientEntityList) }
             }
         }
+
+    override suspend fun registerToken(body: PushTokenBody): Either<NetworkFailure, Unit> = clientRemoteRepository.registerToken(body)
+
+    override fun persistNotificationToken(token: String, transport: String): Either<CoreFailure, Unit> =
+        wrapStorageRequest { tokenStorage.saveToken(NotificationTokenEntity(token, transport)) }
+
+    override fun getNotificationToken(): Either<CoreFailure, NotificationTokenEntity> {
+        return tokenStorage.getToken()?.let { notificationTokenEntity ->
+            Either.Right(notificationTokenEntity)
+        } ?: Either.Left(CoreFailure.MissingClientRegistration)
+    }
```

Comment:
One last suggestion, since a missing token has nothing to do with MissingClientRegistration 
```suggestion
    override fun getNotificationToken(): Either<StorageFailure, NotificationTokenEntity> = wrapStorageRequest { tokenStorage.getToken() `}`
```